modules/genomic_browser/php/endpoints/methylation.class.inc

```diff
@@ -0,0 +1,548 @@
+                     null => 'Any',
+                     '1'  => 'Yes',
+                    )
+                ),
+                'HMM_Island'       => $this->_formElementObjectCreator(
+                    'HMM_Island',
+                    'select',
+                    'HMM Island',
+                    array(
+                     null       => 'Any',
+                     'NULL'     => 'No',
+                     'NOT_NULL' => 'Yes',
+                    )
+                ),
+                'Reg_Feature_Loc'  => $this->_formElementObjectCreator(
+                    'Reg_Feature_Loc',
+                    'text',
+                    'Reg Feature Loc',
+                    null
+                ),
+                'Reg_Feature_Grp'  => $this->_formElementObjectCreator(
+                    'Reg_Feature_Grp',
+                    'select',
+                    'Regulatory feat.:',
+                    $this->getDistinctValues(
+                        'genomic_cpg_annotation',
+                        'reg_feature_group'
+                    )
+                ),
+                'DHS'              => $this->_formElementObjectCreator(
+                    'DHS',
+                    'select',
+                    'DHS',
+                    array(
+                     null   => 'Any',
+                     '1'    => 'Yes',
+                     'NULL' => 'No',
+                    )
+                ),
+                'Platform'         => $this->_formElementObjectCreator(
+                    'Platform',
+                    'select',
+                    'Platform',
+                    $this->getDistinctValues(
+                        'genotyping_platform',
+                        'Name'
+                    )
+                ),
+               );
+    }
+
+    /**
+     * Provide an array of distinct value from a column in
+     * the specified table. The first element has a null key and
+     * its value is 'Any'.
+     *
+     * @param string $table  The tables name
+     * @param string $column The column name
+     *
+     * @return array An array where value are keys as well as values
+     */
+    function getDistinctValues( $table, $column)
+    {
+        $DB      = \Database::singleton();
+        $results = $DB->pselect(
+            "SELECT DISTINCT $column FROM $table ",
+            array()
+        );
+        return array_reduce(
+            $results,
+            function ($carry, $item) {
+                $value = array_values($item)[0];
+                if (!empty($value)) {
+                    $carry[$value] = $value;
+                }
+                return $carry;
+            },
+            array(null => 'Any')
+        );
+
+    }
+
+    /**
+     * Permission verification.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
```

Comment:
Similar to my above comment, maybe this function can be moved to the parent class and called with parameters instead of rewritten in each child class.

Reply:
@johnsaigle I think for this function it's best to not move because the parent class will be used for all modules and where the permission will be different.